File: 2_day/NSOperation_Basic/NSOperation_Basic/ViewController.m
//
//  ViewController.m
//  NSOperation_Basic
//
//  Created by 张正雨 on 2017/5/22.
//  Copyright © 2017年 张正雨. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}


- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event {
    //创建队列  (默认创建的是并行队列)
    NSOperationQueue *queue = [[NSOperationQueue alloc]init];
    
    //GCD队列形式
    // 1、全局  2、主队列  3、串行队列（自己创建）   4、并行队列（自己创建）
    // NSOperationQueue队列形式
    // 1、主队列   2、自己创建的
    
    //创建任务
    NSInvocationOperation *op = [[NSInvocationOperation alloc]initWithTarget:self selector:@selector(operation) object:nil];
    NSInvocationOperation *op1 = [[NSInvocationOperation alloc]initWithTarget:self selector:@selector(operation1) object:nil];
    //将任务放置在队列中  默认调用 start
    [queue addOperation:op];
    [queue addOperation:op1];
    
    [[NSBlockOperation blockOperationWithBlock:^{
        NSLog(@"---3---%@",[NSThread currentThread]);
    }] start];
}
- (void)operation{
    NSLog(@"---1---%@",[NSThread currentThread]);
}
- (void)operation1{
    NSLog(@"---2---%@",[NSThread currentThread]);
}
- (void)operation2{
    NSLog(@"---3---%@",[NSThread currentThread]);
}
@end

File: 1_day/viewController/viewController/ViewController.swift
//
//  ViewController.swift
//  viewController
//
//  Created by 张正雨 on 2017/5/18.
//  Copyright © 2017年 张正雨. All rights reserved.
//

import UIKit

class ViewController: UICollectionViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
        return 50
    }
    override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "cellid", for: indexPath)
        cell.backgroundColor = #colorLiteral(red: 0, green: 0.4784313725, blue: 1, alpha: 1)
        return cell
    }
}

